README
=head1 NAME

re::engine::GNU - Perl extension for blah blah blah

=head1 SYNOPSIS

  use re::engine::GNU;
  blah blah blah

=head1 DESCRIPTION

Stub documentation for re::engine::GNU, created by h2xs. It looks like the
author of the extension was negligent enough to leave the stub
unedited.

Blah blah blah.

=head2 EXPORT

None by default.



=head1 SEE ALSO

Mention other useful documentation such as the documentation of
related modules or operating system documentation (such as man pages
in UNIX), or any relevant external documentation such as RFCs or
standards.

If you have a mailing list set up for your module, mention it here.

If you have a web site set up for your module, mention it here.

=head1 AUTHOR

A. U. Thor, E<lt>a.u.thor@a.galaxy.far.far.awayE<gt>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2015 by A. U. Thor

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.20.1 or,
at your option, any later version of Perl 5 you may have available.




Changes
Changelog for re-engine-GNU

0.026 2023-01-23T20:51:40
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - workflows/dzil-build-and-test.yml: dev+
 - workflows/dzil-build-and-test.yml: Remove AUTHOR_TESTING and
   RELEASE_TESTING when testing the distribution
 - workflows/dzil-build-and-test.yml: Set PAUSE identity bis
 - workflows/dzil-build-and-test.yml: Set PAUSE identity
 - workflows/ci.yml -> workflows/dzil-build-and-test.yml
 - workflows/ci.yml: Trying to understand why dzil test --verbose does not
   work on windows workflow...
 - workflows/ci.yml: use perl -MDist::Zilla::App instead of dzil directly
   for windows workflow
 - engine/GNU.pm: Add a section HOW TO CONTRIBUTE
 - dist.ini: Move AUTHOR_TESTING=1 EXTENDED_TESTING=1 RELEASE_TESTING=1 to
   the run instead of the full workflow - this is causing
   List::SomeUtils::XS to fail because AUTHOR_TESTING sets -Werror
   compilation flag that makes clang-13 bail
 - dist.ini: Use cpan to install (and see why it fails on macos)
   List::SomeUtils::XS
 - dist.ini: Removed non-needed git identity, explicit install of
   Test::Perl::Critic (cpanm bug on MacOS ?)
 - Commit dzil regenerated files
 - dist.ini: Generated Changes from git
 - workflows/ci.yml: Run dzil test with --verbose option
 - workflows/ci.yml: Try to fix git identity
 - workflows/ci.yml: dev+
 - engine/GNU.pm: Dummy commit to get actions to run (!?)
 - workflows/ci.yml: Removed a comment
 - workflows/ci.yml: Fetch repo history (no that big fortunately)
 - workflows/ci.yml: Set git identity
 - workflows/ci.yml:  ~ issue
 - engine/GNU.pm: ## no critic when getting $version
 - workflows/ci.yml: ~/.pause file tentative fix
 - workflows/ci.yml: Create a dummy .pause file
 - re/engine/GNU.pm: Bootstrap even when VERSION is not defined
 - workflows/ci.yml: fix syntax
 - workflows/ci.yml: testing github workflow
 - dzil regenerate(d)
 - dist.ini: Add cpanfile
 - dist.ini: Add Changes to Git::Check.allow_dirty
 - Makefile.PL: copied from release
 - dist.ini: Revisit following comments in #6
 - t/re-engine-GNU.t: explicit show array and hash forms
 - weaver.ini
 - weaver.ini: Initial commit

0.025 2023-01-19T18:30:45Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - engine/GNU.pm: =for Pod::Coverage
 - dist.ini: Remove Test::Vars (unmaintained?), Test::DistManifest (not ok
   for us)
 - engine/GNU.pm: no critic
 - dist.ini: Added Pod::Weaver::PluginBundle::RJBS to develop requires
 - ppport.h: Upgrade to version 3.68
 - GNU.xs: Removed #ifdef REGEXP_LOGICAL_NPARENS_SET, setter macro is a
   no-op if engine do not have logical_nparens member

 [Yves Orton <demerphq@gmail.com>]
 - Update to support and logical_nparens on 5.37.7 and later

0.024 2017-08-01T12:49:20Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - Use SvCANCOW() - fixes #5

0.023 2017-07-31T05:48:54Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - Switch back to perl 5.10 method to set capture string

0.022 2017-07-31T05:03:44Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - Removed offending typecast - closes #4 - RT122641

0.021 2016-01-22T19:04:23Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - dist.ini: Add generated files to clean rule
 - dist.init: Closes #2
 - Fix propagation of newline_anchor in GNU_dupe

0.020 2016-01-21T21:46:00Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - regex_engine.rxfree instead of regex_engine.free - RT111311: Closes #3

0.019 2015-04-12T11:36:39Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - dist.ini: the correct fix for newSVpvn_flags
 - Revert "GNU.xs: perl-5.10.0 newSVpvn_flags"
 - GNU.xs: perl-5.10.0 newSVpvn_flags

0.018 2015-04-04T08:49:04Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - GNU.xs: fixed last parameter (range is a number of characters, bytes I
   believe)

0.017 2015-04-04T08:05:24Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - GNU.xs: removed the hardcoded NULL character in stringification

0.016 2015-04-04T05:32:04Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - mbrtowc: make sure that wcs buffer is always large enough
 - More kWalitee and fix Config::AutoConf hardcoded version dependency
 - Revert "perl-5.10 portability and newSVpvs"
 - perl-5.10 portability and newSVpvs

0.015 2015-04-03T04:00:25Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - GNU.pm: a note about "[.something.]" and "[=something=]
 - regex_internal.c: looks like a bug in original implementation -;
 - GNU.xs: cosmetic changes
 - remove threads::shared, not needed anymore
 - re_string_elem_size_at fix
 - almost working version -;
 - dev+
 - tolower/toupper
 - re_string_elem_size_at: dev+
 - Propagate SV as a helper
 - regex_internal.h: removed (wrong) code
 - regex_internal.c: fix MB_LEN_MAX
 - regex_internal.h: fix octdump prototype
 - portability compiler issues
 - use Config::AutoConf 0.311

0.014 2015-03-28T08:00:49Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - In theory no need anymore for msvcprt.lib
 - Perl portability issues
 - UTF8 fixes
 - POD: revised NOTES section
 - Add -DNDEBUG
 - Fix parameters to re_search
 - dev+
 - __isascii
 - utf8 various fixes
 - Perl_internal_state
 - utf8n_to_uvchr and NUL byte
 - mb stuff
 - Perl mb proxy dev+

0.013 2015-03-21T21:17:16Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - GNU.pm: fix POD
 - removal of libc_calls.c
 - .
 - README.GNU.txt
 - threaded perl support
 - GNU.xs: added missing *THX_

0.012 2015-03-21T14:01:50Z
 [Jean-Damien Durand <jeandamiendurand@free.fr>]
 - cop* stuff removal
 - GNU.xs: fix memory leak
 - general fix with offsets
 - regex_internal.h; handle bool conflict with perl
 - Fix access to rx with MATCH macros and double-free
 - Fix memory errors
 - regcomp.c: warnings fix
 - regcomp.orig.c
 - dev+
 - use threads::shared
 - lock macros
 - aTHX_
 - back to pure xs
 - compile regex.c directly in the XS to profit from perl macros
 - generic calls to libc regardless if in XS or not
 - Moving the NOT_IN_libc model
 - GNU.xs
 - Original regex sources
 - debug+

Makefile.PL
# This Makefile.PL for re-engine-GNU was generated by
# Dist::Zilla::Plugin::MakeMaker::Awesome 0.49.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

use 5.010000;
use ExtUtils::MakeMaker;

use File::Spec;

my %WriteMakefileArgs = (
  "ABSTRACT" => "GNU Regular Expression Engine",
  "AUTHOR" => "Jean-Damien Durand <jeandamiendurand\@free.fr>",
  "CONFIGURE_REQUIRES" => {
    "Capture::Tiny" => 0,
    "Config::AutoConf" => "0.311",
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "POSIX" => 0,
    "diagnostics" => 0,
    "strict" => 0
  },
  "DISTNAME" => "re-engine-GNU",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "re::engine::GNU",
  "PREREQ_PM" => {
    "XSLoader" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0,
    "Thread" => 0
  },
  "VERSION" => "0.026",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

%WriteMakefileArgs = (
    %WriteMakefileArgs,
    OBJECT => 'GNU$(OBJ_EXT)',
    DEFINE => '-DNDEBUG -DNEED_newSVpvn_flags',
    INC => '-I'.File::Spec->curdir,
    clean => { FILES => "config_autoconf.h config_autoconf.log config_REGEXP.h config_REGEXP_wrapped.h config_REGEXP_wrapped.log" },
);

my %FallbackPrereqs = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Test::More" => 0,
  "Thread" => 0,
  "XSLoader" => 0,
  "strict" => 0,
  "warnings" => 0
);

# inserted by Dist::Zilla::Plugin::DynamicPrereqs 0.040
use Config;
if (! $Config{usethreads}) {
  foreach (qw/PREREQ_PM BUILD_REQUIRES TEST_REQUIRES/) {
    delete($WriteMakefileArgs{$_}{Thread}) if (exists($WriteMakefileArgs{$_}));
    delete($FallbackPrereqs{$_}{Thread})   if (exists($FallbackPrereqs{$_}));
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION('6.63_03') } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

package MY;
sub postamble {
    my $self = shift;
    my $config_autoconf = File::Spec->catfile('etc', 'config_autoconf.pl');
    my $config_REGEXP = File::Spec->catfile('etc', 'config_REGEXP.pl');
    return $self->SUPER::postamble . "\nconfig_autoconf.h:\n\t\$(PERLRUN) $config_autoconf\n\nconfig_REGEXP.h:\n\t\$(PERLRUN) $config_REGEXP\n\nGNU\$(OBJ_EXT): config_autoconf.h config_REGEXP.h\n";
}
